Backup data on partitions before resizing them, and restore the data …

daid opened the pull request:
…if the partition was wiped. This backs up to /tmp/, if tmpfs is full, then the partition data that does not fit will be ignored. EMP-401

daid committed:
Backup data on partitions before resizing them, and restore the data if the partition was wiped. This backs up to /tmp/, if tmpfs is full, then the partition data that does not fit will be ignored. EMP-401

daid committed:
Update the backup and restore so it actually works.

daid committed:
Fix the backup restore not working properly.

raysiudak commented on scripts/30_prepare_disk.sh:
```diff
@@ -208,6 +208,51 @@ partition_resize()
+        if mount "${disk_device}" "${temp_mount_dir}"; then
+            mkdir -p "$(dirname "${backup_location}")"
+            if ! tar -czf "${backup_location}" -C "${temp_mount_dir}" .; then
+                rm "${backup_location}"
```
I think you should redirect the output to /dev/null not to pollute the output data and echo that the backup for disk device partition x was successful. 
Same goes for restore.

raysiudak commented on scripts/30_prepare_disk.sh:
```diff
@@ -208,6 +208,51 @@ partition_resize()
+
+    sfdisk --quiet --dump "${TARGET_STORAGE_DEVICE}" | \
+    while IFS="${IFS}:=," read -r disk_device _ disk_start _ disk_size _; do
+        if grep -q "${disk_device}" /proc/mounts; then
```
Shellcheck won't complain, but i but it is more consistent to also add quotes to "/proc/mount", same goes for the restore.

raysiudak reviewed (changes requested):
In general, for the next version we should optimize it, not everything needs to be copied or restored. And then we could create a 25_backup.sh and a 35_restore.sh or 55_restore.sh script.

raysiudak commented on scripts/30_prepare_disk.sh:
```diff
@@ -208,6 +208,51 @@ partition_resize()
+        if [ -f "${backup_file}" ]; then
+            mount "${disk_device}" "${temp_mount_dir}"
+            # We only restore if the parition looks empty. There can be a lost+found on an empty partition, so ignore that.
+            if [ "$(ls "${temp_mount_dir}" | wc -l)" < 2; then
```
Shellcheck SC2012: Use find instead of ls to better handle non-alphanumeric filenames.

raysiudak committed:
Fixed shellcheck error SC2012 and made some small improvements

- Add some empty lines for readability
- Fixed shellcheck error: SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
- Redirect tar output to prevent pollution
- Add quotes around variables to satisfy shellcheck

Signed-off-by: Raymond Siudak <r.siudak@ultimaker.com>
